### ISSUE TITLE
[RxJava 2.0.0-RC2] Observable.delay doesn't respect the provided Scheduler

### ISSUE THREAD

**Issue by nsk-mironov** [ReactiveX/RxJava#4521]
``` Java
public class DelayBug {
  public static void main(String[] args) throws Exception {
    final CountDownLatch latch = new CountDownLatch(1);

    Observable.<String>error(new Exception())
        .delay(100, TimeUnit.MILLISECONDS, Schedulers.io())
        .doOnError(throwable -> System.out.println("onError @ " + Thread.currentThread().getName()))
        .onErrorResumeNext(Observable.empty())
        .doAfterTerminate(latch::countDown)
        .subscribe();

    latch.await();
  }
}
```

Output:

```
onError @ main
```

Expected output:

```
onError @ RxCachedThreadScheduler-1
```

**Comment by akarnokd** [comment]
Please check the available overloads for the feature you are looking for.

**Comment by nsk-mironov** [comment]
I don't want to delay the error, but I still want to receive `onError` callback on the provided scheduler. This is how RxJava 1 works ([OperatorDelay.java#L66-L78](https://github.com/ReactiveX/RxJava/blob/1.x/src/main/java/rx/internal/operators/OperatorDelay.java#L66-L78)). I think the following code [ObservableDelay.java#L89-L105](https://github.com/ReactiveX/RxJava/blob/2.x/src/main/java/io/reactivex/internal/operators/observable/ObservableDelay.java#L89-L105) should be:

``` java
@Override
public void onError(final Throwable t) {
    w.schedule(new Runnable() {
        @Override
        public void run() {
            try {
                actual.onError(t);
            } finally {
                w.dispose();
            }
        }
    }, delayError ? delay : 0, unit);
}
```

**Comment by akarnokd** [comment]
PR welcome.

**Comment by nsk-mironov** [comment]
Closed via #4522
